components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java

```diff
@@ -166,6 +169,20 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response)
                 // User denied logout.
                 redirectURL = OIDCSessionManagementUtil
                         .getErrorPageURL(OAuth2ErrorCodes.ACCESS_DENIED, "End User denied the logout request");
+
+                // Set postLogoutRedirectUri as redirectURL.
+                boolean postLogoutRedirectUriRedirectIsEnabled =
+                        Boolean.parseBoolean(IdentityUtil.getProperty(OIDC_LOGOUT_CONSENT_DENIAL_REDIRECT_URL));
+                if (postLogoutRedirectUriRedirectIsEnabled) {
```

Comment:
better to make this a private method with good name to make the entire code more readable

Reply:
okay